monai/metrics/morphological_hausdorff_distance.py

```diff
@@ -0,0 +1,85 @@
+# limitations under the License.
+
+import warnings
+from typing import Optional, Union
+
+import numpy as np
+import torch
+from monai._extensions.loader import load_module
+from .metric import CumulativeIterationMetric
+
+__all__ = ["MorphologicalHausdorffDistanceMetric"]
+
+
+class MorphologicalHausdorffDistanceMetric(CumulativeIterationMetric):
+    """
+    Work is based onthe principle of application of  mathemathical morphology more precisely dilatation, more details is presented in works [1] and [2]. Calculated  is roughly related to the largest
+thickness of the difference between the true and estimatedmasks, and constitutes Approximation of true Hausdorff distance. The correlation between true and morphological Hausdorf distance is around 0.9, less for robust version .
+Hovewer this implementation is also around 100 times faster than exact MONAI Hausdorff distance calculation. One can measure 3 diffrent metrics based on Hausdorff ditance.
+
+    1)simple Hausdorff distance - it tell what is the distance between two most distant points one from y and other from y_pred
+
+    2)robust Hausdorff distance - modification of 1) where we stop analyzing points when we already analyzed given percent of total number of points - is less sensitive to outliers. It is also slightly faster
+
+    3)mean Hausdorff distance - can be understood as approximately mean distance between pair of points where one point is from y and other from y_pred.
+
+    4)additionally one can request as a result one dimensional tensor that will contain data for each point what is HD value for it (like we would do Hausdorff distance calculation only for this point and all points from other mask)
+
+    Compute Hausdorff Distance between two tensors. In addition, specify the `percent` (for robust Hausdorff distance )
+    parameter can get the percentile of the distance. Input `y_pred` is compared with ground truth `y`.
+
+    Args:
+        percent: an optional float number between 0 and 1. If specified, the corresponding
+            percent of the Hausdorff Distance rather than the maximum result will be achieved.
+            Defaults to 1.0 .
+
+    1) Érick Oliveira Rodrigues,An efficient and locality-oriented Hausdorff distance algorithm: Proposal and analysis of paradigms and implementations, Pattern Recognition,Volume 117,2021
+    2) Karimi, Davood & Salcudean, Septimiu. (2019). Reducing the Hausdorff Distance in Medical Image Segmentation With Convolutional Neural Networks. IEEE Transactions on Medical Imaging. PP. 1-1. 10.1109/TMI.2019.2930068.
+
+    """
+
+    def __init__(
+        self,
+        percent: float = 1.0,
+    ) -> None:
+        super().__init__()
+        self.percent = percent
+        self.compiled_extension = load_module(
+            "hausdorff_cpp"
+        )
+
+
+    def compute_morphological_hausdorff_distance(
+        self,
+        y_pred:  torch.Tensor,
+        y: torch.Tensor,
+        numberToLookFor: torch.Tensor):
```

Comment:
We're sticking to Python naming wherever we can so please change this to `number_to_look_for` or maybe something shorter like `compare_values`.

Reply:
thanks!